daffodil-tdml-processor/src/test/resources/test/tdml/testTDMLErrorsWarningsMatchAttribute.tdml

```diff
@@ -0,0 +1,150 @@
+
+    <tdml:validationErrors match="none">
+      <tdml:error>pattern facet</tdml:error>
+      <tdml:error>WILL NOT MATCH ON ANY</tdml:error>
+    </tdml:validationErrors>
+
+  </tdml:parserTestCase>
+
+<tdml:defineSchema name="causesErrors" elementFormDefault="unqualified">
+
+  <xs:include schemaLocation="/org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd"/>
+    <dfdl:format ref="ex:GeneralFormat" lengthKind="explicit" occursCountKind="implicit"/>
+
+  <xs:element name="causesUnparseError" dfdl:lengthKind="implicit">
+    <xs:complexType>
+      <xs:sequence>
+        <xs:element name="A" type="xs:string" dfdl:length="1" />
+        <xs:element name="AmbigElt" type="xs:string" dfdl:length="1" />
+        <xs:element name="B" type="xs:string" dfdl:length="1" />
+        <xs:element name="AmbigElt" type="xs:string" dfdl:length="1" minOccurs="0" />
+        <xs:element name="UnparseFails" type="xs:string" dfdl:length="1"
+           dfdl:outputValueCalc="{ ../AmbigElt }"/>
+      </xs:sequence>
+    </xs:complexType>
+  </xs:element>
+
+  <xs:element name="causesValidationError" dfdl:lengthKind="implicit">
+    <xs:complexType>
+      <xs:sequence>
+        <xs:element name="A" dfdl:length="1" dfdl:inputValueCalc="{ 'A' }">
+          <xs:simpleType>
+            <xs:restriction base="xs:string">
+              <xs:pattern value="[^A-FH-OQ-Z]"/>
+            </xs:restriction>
+          </xs:simpleType>
+        </xs:element>
+      </xs:sequence>
+    </xs:complexType>
+  </xs:element>
+
+</tdml:defineSchema>
+
+<tdml:unparserTestCase name="getUnparserWarningWhenExpectingError"
+                       root="causesUnparseError" model="causesErrors">
+  <tdml:infoset>
+    <tdml:dfdlInfoset>
+      <ex:causesUnparseError>
+        <A>1</A>
+        <AmbigElt>2</AmbigElt>
+        <B>3</B>
+        <AmbigElt>4</AmbigElt>
+      </ex:causesUnparseError>
+    </tdml:dfdlInfoset>
+  </tdml:infoset>
+  
+  <tdml:document><![CDATA[12344]]></tdml:document>
+
+  <tdml:errors match="any">
+    <tdml:error>WILL NOT MATCH ON ANY</tdml:error>
+    <tdml:error>Schema Definition Error</tdml:error>
+  </tdml:errors>
+
+  <tdml:warnings match="any">
+    <tdml:warning>WILL NOT MATCH ON ANY</tdml:warning>
+    <tdml:warning>Schema Definition Warning</tdml:warning>
+    <tdml:warning>AmbigElt</tdml:warning>
+    <tdml:warning>query-style</tdml:warning>
+  </tdml:warnings>
+</tdml:unparserTestCase>
+
+
+<tdml:parserTestCase name="expectsAnyValidationError" root="causesValidationError"
+                     model="causesErrors" validation="on">
+
+  <tdml:document/>
+
+  <tdml:infoset>
+    <tdml:dfdlInfoset>
+      <ex:causesValidationError>
+        <A>A</A>
+      </ex:causesValidationError>
+    </tdml:dfdlInfoset>
+  </tdml:infoset>
+
+  <tdml:validationErrors match="any">
+    <tdml:error>value 'A' of element 'A' is not valid</tdml:error>
+    <tdml:error>WILL NOT MATCH ON ANY</tdml:error>
+  </tdml:validationErrors>
+</tdml:parserTestCase>
```

Comment:
Does this support a way to say that no diagnostics of a type should exist? Or maybe a way to say there must be warnings or errors, but that we don't necessarily care what they are?

Maybe an empty tag with match="all" means an diag type must exist, but we don't care about the message.
```xml
<tdml:errors match="all" />
```

And a empty tag with match="none" means a diag type must not exist:
```xml
<tdml:warnings match="none" />
```

I'm not sure if that logic makes sense or if it should be reversed, but something along those lines might be useful.